app/src/main/java/com/asfoundation/wallet/util/CurrencyFormatUtils.kt

```diff
@@ -0,0 +1,106 @@
+package com.asfoundation.wallet.util
+
+import com.asfoundation.wallet.ui.iab.FiatValue
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.text.NumberFormat
+
+class CurrencyFormatUtils {
```

Comment:
create a method that receives a bigdecimal and currency type.
  ```
fun formatCurrency(value: BigDecimal, currencyType: WalletCurrency): String {
    return formatCurrency(value.toDouble(), currencyType)
  }
```
This will make you save calling toDouble() every time:  
`val appcValue = formatter.formatCurrency(appcAmount, WalletCurrency.APPCOINS)`
instead of 
`val appcValue = formatter.formatCurrency(appcAmount.toDouble(), WalletCurrency.APPCOINS)`